ui/src/tag/suggest.ts

```diff
@@ -21,10 +23,17 @@ export const useSuggest = (
         skip: exactMatch === undefined || skipValue,
     });
 
-    if (exactMatch && tagValue !== undefined && usedTags.indexOf(exactMatch.key) === -1 && !skipValue) {
-        return suggestTagValue(exactMatch, tagValue, valueResult);
+    let usedKeys: string[] = [];
+    if (!allowDuplicateKeys) {
+        usedKeys = usedTags.map((t) => t.tag.key);
+    }
```

Comment:
usedKeys shouldn't be empty when allowDuplicateKeys isset. Can this be checked later. E.g.
```
    // usedTags is usedKeys in suggestTag.
    if (usedTags.indexOf(tagKey) !== -1 && !allowDuplicateKeys) {
        availableTags = [specialTag(tagKey, 'used'), ...availableTags];
    }
```
This makes it more obvious, what should happen.